Fix skin fail sound continuing to play after leaving gameplay

shinobu-uwu opened the pull request:
Resolves #22173

shinobu-uwu committed:
Fix failSample still playing after player left FailOverlay

peppy commented on osu.Game/Screens/Play/Player.cs:
```diff
@@ -612,6 +612,8 @@ protected void PerformExit(bool showDialogFirst)
             // if an exit has been requested, cancel any pending completion (the user has shown intention to exit).
             resultsDisplayDelegate?.Cancel();
 
+            failAnimationLayer.StopSample();
```
While this will work, I feel like it should be combined with the `RemoveFilters` call into a new `public` method (with `RemoveFilters` being changed to `private` and called by the new method).

shinobu-uwu committed:
Create StopSampleAndRemoveFilters method and change RemoveFilters to private

shinobu-uwu committed:
Replace RemoveFilters call with StopSampleAndRemoveFilters

shinobu-uwu committed:
Merge branch 'master' into fix-fail-sample-playing-after-leaving-dialog

bdach committed:
Remove unnecessary sample stop in `Restart()`

It is unnecessary, as a successful restart will exit the current player
screen, and `OnExiting()` has another `StopSampleAndRemoveFilters()`
call, which means that in the restart flow the sample was actually
getting stopped twice.

Standard exit flow is fine, it only stopped the sample once.

bdach committed:
Inline `StopSample()` into `StopSampleAndRemoveFilters()`

The first method no longer has any callers except for the second one.

bdach committed:
Rename `Stop{SampleAndRemoveFilters -> }()`

Now that just one method for stopping samples is left, let's just
repurpose st as the general "stop global effects" method rather than
have it there with a hyperspecific name. It also has good symmetry, as
there already was a `Start()` method in the class.

bdach committed:
Merge branch 'master' into fix-fail-sample-playing-after-leaving-dialog

bdach commented:
I have applied a few code clean-ups to this. Rationale for each change is given in commit message.

@ppy/team-client request second review if able as I changed a substantial amount of code relative to the size of the PR.
